### PR TITLE
Server dev

### PR DESCRIPTION
**Pull request by ie208**
Group chat API is now functional:

- **CREATEGROUP** (user will automatically join the group, no need to separately send JOINGROUP request)

- **JOINGROUP** - simply joins a group given the groupname. No request/accept system at the moment.

- **GETALLGROUPS** - returns a list of all the groups in the system, and the members in the groups

- **SEARCHGROUPS** - similar to searching users, searches the groups for a like query and returns a list of groups and their members

- **GROUPTEXT** - send a text message to the group. This will then be sent to all the other users who are members of the group. (whether this text will be encrypted or not, and how etc, depends entirely on whether the clients encrypt the messages or not - as the server does not determine this!). Similar to regular user texts, this will save texts in a message map and send them on login if the user is offline at the time.

Deleting a group, removing a user from a group, blocking a user from entering a group or a request/accept system, and other more advanced functionality is not (yet) implemented.

### CODE REVIEW

File: TeamZeroServer/src/server/DbConnection.java [hunk 1]
```diff
@@ -413,17 +419,108 @@ public boolean addNewGroup(String groupName, String creatingUser) throws SQLExce
...
-	//TODO
-	public ArrayList<Group> getSearchedGroups(String search){
-		return new ArrayList<Group>();
+	
+	public ArrayList<Group> getSearchedGroups(String search) throws SQLException{
+		ArrayList<Group> searchedGroups = new ArrayList<Group>();
+		Connection conn = this.connect();
+		try {
+			
+			//TODO fix this - currently it is not getting the members
```
Comment (EmadNaguiDarwish) [inline TeamZeroServer/src/server/DbConnection.java]: Is it still not getting the members?

Comment (ie208) [reply TeamZeroServer/src/server/DbConnection.java]: Ah, sorry about that. It is getting the members now, I just forgot to take out the comment

Comment (EmadNaguiDarwish) [reply TeamZeroServer/src/server/DbConnection.java]: Ok, no problem. I'll remove this comment.

**Comment by MetallicFocus** [comment]
**For CREATEGROUP** - "_user will automatically join the group, no need to separately send `JOINGROUP` request_" --> I'm adding this part to the Wiki page too

**Pull request merged**